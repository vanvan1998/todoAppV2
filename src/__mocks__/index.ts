const todo = {
  email: 'nguyensivan1998+102@gmail.com',
  id: '82zB2At6gnxs0SbDJ8SW',
  notification: true,
  createdAt: '2024-05-10T01:14:38',
  title: '333333',
  detail: '3333333',
  startTime: '01:00:21',
  startDate: '05/10/2024',
  category: 1,
  completed: false,
  userId: 'j3WklyoTrBg4E1GKNz0Gn0az34V2',
  checkList: []
};
export { todo };
