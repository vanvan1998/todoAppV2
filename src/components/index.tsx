export { AddTodoModal } from './AddTodoModal';
export { Search } from './Search';
export { TodoItem } from './TodoItem';
export { Container } from './Container';
export { Header } from './Header';
export { AddTodo } from './addTodo';
export { SortBy } from './SortBy';
export { Input } from './input';
export { Button } from './button';
