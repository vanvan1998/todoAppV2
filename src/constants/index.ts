export const SORT_BY = {
  NEWEST: 'NEWEST',
  OLDEST: 'OLDEST',
  NONE: 'NONE'
};

export const MODE = {
  UPDATE: 'UPDATE',
  ADD: 'ADD'
};

export const DATE_FORMAT = 'MM/DD/YYYY';
export const TIME_FORMAT = 'HH:mm:ss';
