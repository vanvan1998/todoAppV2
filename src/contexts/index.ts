export * from './AppProvider';
export * from './GlobalLoadingProvider';
export * from './AuthProvider';
