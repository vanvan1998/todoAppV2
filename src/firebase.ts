import { initializeApp } from 'firebase/app';
import { getAuth, signInWithPopup } from 'firebase/auth';
import { getStorage } from 'firebase/storage';
import { getFirestore } from 'firebase/firestore';
import { GoogleAuthProvider } from 'firebase/auth';

export const app = initializeApp({
  apiKey: process.env.NEXT_PUBLIC_FIREBASE_API_KEY,
  authDomain: process.env.NEXT_PUBLIC_FIREBASE_AUTH_DOMAIN,
  databaseURL: process.env.NEXT_PUBLIC_FIREBASE_DATABASE_URL,
  projectId: process.env.NEXT_PUBLIC_FIREBASE_PROJECT_ID,
  storageBucket: process.env.NEXT_PUBLIC_FIREBASE_STORAGE_BUCKET,
  messagingSenderId: process.env.NEXT_PUBLIC_FIREBASE_MESSAGING_SENDER_ID,
  appId: process.env.NEXT_PUBLIC_FIREBASE_APP_ID
});

const provider = new GoogleAuthProvider();
provider.setCustomParameters({ prompt: 'select_account' });

export const auth = getAuth(app);
export const storage = getStorage(app);
export const db = getFirestore(app);

export const signInWithGoogle = () => signInWithPopup(auth, provider);