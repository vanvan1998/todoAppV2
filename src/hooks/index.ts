export * from './useFirebase';
export * from './useNotification';
export * from './useMediaQuery';
