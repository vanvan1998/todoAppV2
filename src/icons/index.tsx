export * from './AccountIcon';
export * from './HomeIcon';
export * from './CheckedIcon';
export * from './UnCheckedIcon';
export * from './EditIcon';
export * from './DeleteIcon';
export * from './ExpandLessIcon';
export * from './ExpandMoreIcon';
export * from './CloseIcon';
export * from './PersonIcon';
export * from './UrgentIcon';
export * from './ImportantIcon';
export * from './UpcomingIcon';
export * from './SignOutIcon';
