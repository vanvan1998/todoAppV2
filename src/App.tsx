import React from 'react';
import { AuthProvider } from './contexts/AuthContext';
import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
import {
  Signup,
  LeftMenu,
  Login,
  PrivateRoute,
  ForgotPassword,
  UpdatePassword,
  UpdateProfile,
  DashBoard
} from './modules';

function App() {
  return (
    <div style={{ minHeight: '100vh', display: 'flex' }}>
      <Router>
        <AuthProvider>
          <Routes>
            <Route
              path='/update-password'
              element={
                <PrivateRoute>
                  <UpdatePassword />
                </PrivateRoute>
              }
            />
            <Route
              path='/update-profile'
              element={
                <PrivateRoute>
                  <UpdateProfile />
                </PrivateRoute>
              }
            />
            <Route path='/signup' element={<Signup />} />
            <Route path='/login' element={<Login />} />
            <Route path='/forgot-password' element={<ForgotPassword />} />
            <Route
              path='/left-menu'
              element={
                <PrivateRoute>
                  <LeftMenu />
                </PrivateRoute>
              }
            />
            <Route
              path='/'
              element={
                <PrivateRoute>
                  <DashBoard />
                </PrivateRoute>
              }
            />
          </Routes>
        </AuthProvider>
      </Router>
    </div>
  );
}

export default App;
