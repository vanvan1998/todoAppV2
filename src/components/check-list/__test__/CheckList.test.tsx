import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import { CheckList } from '..';
import { todo } from 'src/__mocks__';

describe('Test TodoDetailItem component', () => {
  test.only('should match its snapshot', async () => {
    const component = render(
      <CheckList
        todo={{
          ...todo,
          checkList: [
            {
              completed: true,
              title: 'vvvvv'
            }
          ]
        }}
        isMobile
        handleCompleteTodo={jest.fn()}
        handleDeleteTodo={jest.fn()}
        handleUpdateTodo={jest.fn()}
      />
    );

    const checkButton = component.getByTestId('check-button');
    fireEvent.click(checkButton);

    const deleteButton = component.getByTestId('delete-check-button');
    fireEvent.click(deleteButton);

    expect(component).toMatchSnapshot();
  });

  test.only('should handle uncheck checklist', async () => {
    const component = render(
      <CheckList
        todo={{
          ...todo,
          checkList: [
            {
              completed: false,
              title: 'vvvvv'
            }
          ]
        }}
        isMobile
        handleCompleteTodo={jest.fn()}
        handleDeleteTodo={jest.fn()}
        handleUpdateTodo={jest.fn()}
      />
    );

    const checkButton = component.getByTestId('uncheck-button');
    fireEvent.click(checkButton);

    const deleteButton = component.getByTestId('delete-check-button');
    fireEvent.click(deleteButton);

    expect(component).toMatchSnapshot();
  });

  test.only('should render empty checklist todo snapshot', async () => {
    const component = render(
      <CheckList
        todo={todo}
        isMobile={false}
        handleCompleteTodo={jest.fn()}
        handleDeleteTodo={jest.fn()}
        handleUpdateTodo={jest.fn()}
      />
    );

    expect(component).toMatchSnapshot();
  });
});
