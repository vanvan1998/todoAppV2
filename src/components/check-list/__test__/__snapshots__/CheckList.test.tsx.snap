// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test TodoDetailItem component should handle uncheck checklist 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-iqyJx fzwjUx"
      >
        <div
          class="sc-gtLWhw jPWJOW"
          style="font-weight: bold; font-size: 16px; padding-bottom: 6px;"
        >
          Check List
        </div>
        <div
          class="sc-jIyAiq iridzR"
          color="#d25a12"
        >
          <svg
            color="#d25a12"
            data-testid="uncheck-button"
            fill="currentColor"
            height="18"
            stroke="currentColor"
            stroke-width="0"
            version="1.1"
            viewBox="0 0 16 16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14 0h-12c-1.1 0-2 0.9-2 2v12c0 1.1 0.9 2 2 2h12c1.1 0 2-0.9 2-2v-12c0-1.1-0.9-2-2-2zM14 14h-12v-12h12v12z"
            />
          </svg>
          <div
            class="sc-lnsxGb glTJah"
          >
            <div
              class="sc-gtLWhw jPWJOW"
              style="flex: 1;"
            >
              vvvvv
            </div>
            <svg
              data-testid="delete-check-button"
              fill="#e2342d"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m289.94 256 95-95A24 24 0 0 0 351 127l-95 95-95-95a24 24 0 0 0-34 34l95 95-95 95a24 24 0 1 0 34 34l95-95 95 95a24 24 0 0 0 34-34z"
              />
            </svg>
          </div>
        </div>
        <div
          class="sc-eDHQDy iWIXDk"
        >
          <input
            class="sc-giBObj hDMvKn"
            placeholder="Enter new task todo..."
            type="text"
            value=""
          />
          <svg
            color="#d25a12"
            fill="currentColor"
            height="20"
            stroke="currentColor"
            stroke-width="0"
            style="cursor: pointer;"
            viewBox="0 0 512 512"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M416 277.333H277.333V416h-42.666V277.333H96v-42.666h138.667V96h42.666v138.667H416v42.666z"
            />
          </svg>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-iqyJx fzwjUx"
    >
      <div
        class="sc-gtLWhw jPWJOW"
        style="font-weight: bold; font-size: 16px; padding-bottom: 6px;"
      >
        Check List
      </div>
      <div
        class="sc-jIyAiq iridzR"
        color="#d25a12"
      >
        <svg
          color="#d25a12"
          data-testid="uncheck-button"
          fill="currentColor"
          height="18"
          stroke="currentColor"
          stroke-width="0"
          version="1.1"
          viewBox="0 0 16 16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M14 0h-12c-1.1 0-2 0.9-2 2v12c0 1.1 0.9 2 2 2h12c1.1 0 2-0.9 2-2v-12c0-1.1-0.9-2-2-2zM14 14h-12v-12h12v12z"
          />
        </svg>
        <div
          class="sc-lnsxGb glTJah"
        >
          <div
            class="sc-gtLWhw jPWJOW"
            style="flex: 1;"
          >
            vvvvv
          </div>
          <svg
            data-testid="delete-check-button"
            fill="#e2342d"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m289.94 256 95-95A24 24 0 0 0 351 127l-95 95-95-95a24 24 0 0 0-34 34l95 95-95 95a24 24 0 1 0 34 34l95-95 95 95a24 24 0 0 0 34-34z"
            />
          </svg>
        </div>
      </div>
      <div
        class="sc-eDHQDy iWIXDk"
      >
        <input
          class="sc-giBObj hDMvKn"
          placeholder="Enter new task todo..."
          type="text"
          value=""
        />
        <svg
          color="#d25a12"
          fill="currentColor"
          height="20"
          stroke="currentColor"
          stroke-width="0"
          style="cursor: pointer;"
          viewBox="0 0 512 512"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M416 277.333H277.333V416h-42.666V277.333H96v-42.666h138.667V96h42.666v138.667H416v42.666z"
          />
        </svg>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test TodoDetailItem component should match its snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-iqyJx fzwjUx"
      >
        <div
          class="sc-gtLWhw jPWJOW"
          style="font-weight: bold; font-size: 16px; padding-bottom: 6px;"
        >
          Check List
        </div>
        <div
          class="sc-jIyAiq iridzR"
          color="#d25a12"
        >
          <svg
            data-testid="check-button"
            fill="#d25a12"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M64 32C28.7 32 0 60.7 0 96V416c0 35.3 28.7 64 64 64H384c35.3 0 64-28.7 64-64V96c0-35.3-28.7-64-64-64H64zM337 209L209 337c-9.4 9.4-24.6 9.4-33.9 0l-64-64c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.4 33.9 0l47 47L303 175c9.4-9.4 24.6-9.4 33.9 0s9.4 24.6 0 33.9z"
            />
          </svg>
          <div
            class="sc-lnsxGb glTJah"
          >
            <div
              class="sc-gtLWhw jPWJOW"
              style="flex: 1;"
            >
              vvvvv
            </div>
            <svg
              data-testid="delete-check-button"
              fill="#e2342d"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m289.94 256 95-95A24 24 0 0 0 351 127l-95 95-95-95a24 24 0 0 0-34 34l95 95-95 95a24 24 0 1 0 34 34l95-95 95 95a24 24 0 0 0 34-34z"
              />
            </svg>
          </div>
        </div>
        <div
          class="sc-eDHQDy iWIXDk"
        >
          <input
            class="sc-giBObj hDMvKn"
            placeholder="Enter new task todo..."
            type="text"
            value=""
          />
          <svg
            color="#d25a12"
            fill="currentColor"
            height="20"
            stroke="currentColor"
            stroke-width="0"
            style="cursor: pointer;"
            viewBox="0 0 512 512"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M416 277.333H277.333V416h-42.666V277.333H96v-42.666h138.667V96h42.666v138.667H416v42.666z"
            />
          </svg>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-iqyJx fzwjUx"
    >
      <div
        class="sc-gtLWhw jPWJOW"
        style="font-weight: bold; font-size: 16px; padding-bottom: 6px;"
      >
        Check List
      </div>
      <div
        class="sc-jIyAiq iridzR"
        color="#d25a12"
      >
        <svg
          data-testid="check-button"
          fill="#d25a12"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 448 512"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M64 32C28.7 32 0 60.7 0 96V416c0 35.3 28.7 64 64 64H384c35.3 0 64-28.7 64-64V96c0-35.3-28.7-64-64-64H64zM337 209L209 337c-9.4 9.4-24.6 9.4-33.9 0l-64-64c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.4 33.9 0l47 47L303 175c9.4-9.4 24.6-9.4 33.9 0s9.4 24.6 0 33.9z"
          />
        </svg>
        <div
          class="sc-lnsxGb glTJah"
        >
          <div
            class="sc-gtLWhw jPWJOW"
            style="flex: 1;"
          >
            vvvvv
          </div>
          <svg
            data-testid="delete-check-button"
            fill="#e2342d"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m289.94 256 95-95A24 24 0 0 0 351 127l-95 95-95-95a24 24 0 0 0-34 34l95 95-95 95a24 24 0 1 0 34 34l95-95 95 95a24 24 0 0 0 34-34z"
            />
          </svg>
        </div>
      </div>
      <div
        class="sc-eDHQDy iWIXDk"
      >
        <input
          class="sc-giBObj hDMvKn"
          placeholder="Enter new task todo..."
          type="text"
          value=""
        />
        <svg
          color="#d25a12"
          fill="currentColor"
          height="20"
          stroke="currentColor"
          stroke-width="0"
          style="cursor: pointer;"
          viewBox="0 0 512 512"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M416 277.333H277.333V416h-42.666V277.333H96v-42.666h138.667V96h42.666v138.667H416v42.666z"
          />
        </svg>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test TodoDetailItem component should render empty checklist todo snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-iqyJx ilDgpx"
      >
        <div
          class="sc-gtLWhw jPWJOW"
          style="font-weight: bold; font-size: 16px; padding-bottom: 6px;"
        >
          Check List
        </div>
        <div
          class="sc-eDHQDy iWIXDk"
        >
          <input
            class="sc-giBObj hDMvKn"
            placeholder="Enter new task todo..."
            type="text"
            value=""
          />
          <svg
            color="#d25a12"
            fill="currentColor"
            height="20"
            stroke="currentColor"
            stroke-width="0"
            style="cursor: pointer;"
            viewBox="0 0 512 512"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M416 277.333H277.333V416h-42.666V277.333H96v-42.666h138.667V96h42.666v138.667H416v42.666z"
            />
          </svg>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-iqyJx ilDgpx"
    >
      <div
        class="sc-gtLWhw jPWJOW"
        style="font-weight: bold; font-size: 16px; padding-bottom: 6px;"
      >
        Check List
      </div>
      <div
        class="sc-eDHQDy iWIXDk"
      >
        <input
          class="sc-giBObj hDMvKn"
          placeholder="Enter new task todo..."
          type="text"
          value=""
        />
        <svg
          color="#d25a12"
          fill="currentColor"
          height="20"
          stroke="currentColor"
          stroke-width="0"
          style="cursor: pointer;"
          viewBox="0 0 512 512"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M416 277.333H277.333V416h-42.666V277.333H96v-42.666h138.667V96h42.666v138.667H416v42.666z"
          />
        </svg>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
